akka-stream-tests/src/test/scala/akka/stream/scaladsl/MaybeSourceSpec.scala

```diff
@@ -35,6 +35,15 @@ class MaybeSourceSpec extends StreamSpec with DefaultTimeout {
       f.future.futureValue shouldEqual None
     }
 
+    "complete materialized promise with None when stream cancels with a failure cause" in assertAllStagesStopped {
+      val (promise, killswitch) = Source.maybe[Int].viaMat(KillSwitches.single)(Keep.both).to(Sink.ignore).run()
```

Comment:
Isn't it somewhat weird anyway, that this promise is tried to be used for two-way communication?

Reply:
I agree, I don't want to change that now though.

Reply:
Agreed.